parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -403,10 +403,20 @@ function nodePrep {
     elif [ "${GPU_NODE}" = "true" ] && [ "${skip_nvidia_driver_install}" != "true" ] && [ "${ENABLE_MANAGED_GPU_EXPERIENCE}" = "true" ]; then
         logs_to_events "AKS.CSE.installNvidiaManagedExpPkgFromCache" "installNvidiaManagedExpPkgFromCache" || exit $ERR_NVIDIA_DCGM_INSTALL
         logs_to_events "AKS.CSE.startNvidiaManagedExpServices" "startNvidiaManagedExpServices" || exit $ERR_NVIDIA_DCGM_EXPORTER_FAIL
-    elif [ "${GPU_NODE}" = "true" ] && [ "${skip_nvidia_driver_install}" != "true" ] && [ "${ENABLE_MANAGED_GPU_EXPERIENCE}" = "false" ]; then
+
+        # Add node label to indicate DCGM exporter is enabled
+        DCGM_EXPORTER_LABEL="kubernetes.azure.com/dcgm-exporter=enabled"
+        echo "adding node label $DCGM_EXPORTER_LABEL"
+        addKubeletNodeLabel $DCGM_EXPORTER_LABEL
```

Comment:
would we need to ensure that ensure Kubelet is call after ?  since this is part of `nodePrep` If the start of Kubelet occurs after we would not have those labels correct ?